### ISSUE TITLE
Problem with connection. Failed to establish SSL connection with the HTTP server

### ISSUE THREAD

**Issue by xerasy** [X-rus/xNet#54]
```
request.UserAgent = useragent;
request.IgnoreProtocolErrors = true;
request.SslCertificateValidatorCallback += (sender, certificate, chain, sslPolicyErrors) => true;

var result = request.Get("https://sscasino.online/");
```
but i get error 
`Failed to establish SSL connection with the HTTP server`

i get the same the error for WebRequest 
```
  WebRequest req = WebRequest.Create(Url);
  WebResponse resp = req.GetResponse();
  Stream stream = resp.GetResponseStream();
  StreamReader sr = new StreamReader(stream);
  string Out = sr.ReadToEnd();
  sr.Close();
```
but solution for WebRequest is a add line
`ServicePointManager.SecurityProtocol = SecurityProtocolType.Tls12;`

what is the solution for xnet ?

**Comment by gzlance** [comment]
sslStream.AuthenticateAsClient(address.Host);
修改成如下：
sslStream.AuthenticateAsClient(address.Host, new X509Store(StoreName.Root).Certificates, SslProtocols.Ssl3 | SslProtocols.Ssl2 | SslProtocols.Tls, false);

这样应该能解决你的问题。

**Comment by Fedorus** [comment]
Looks like its a SslStream issue.
Put in on HttpRequests.cs on line ~2595
sslStream.AuthenticateAsClient(address.Host, null, SslProtocols.Tls12 | SslProtocols.Tls11 | SslProtocols.Default, false);

instead default line
sslStream.AuthenticateAsClient(...)

**Comment by whynotme8998** [comment]
Yes! you must set "request.SslCertificateValidatorCallback += delegate" return true

**Comment by angelocart** [comment]
Replacing the line `sslStream.AuthenticateAsClient(...)` **with** `sslStream.AuthenticateAsClient(address.Host, null, SslProtocols.Tls12, false);` returns an error of "SslProtocols' does not contain a definition for 'Tls12'.

Please fix, thanks!

**Comment by Fedorus** [comment]
> Replacing the line `sslStream.AuthenticateAsClient(...)` **with** `sslStream.AuthenticateAsClient(address.Host, null, SslProtocols.Tls12, false);` returns an error of "SslProtocols' does not contain a definition for 'Tls12'.
> 
> Please fix, thanks!

This repository no more supported. Try this one:
https://github.com/Fedorus/xNetStandard

Its for .netStandart. Lost all operations connected with winAPI plus GZip connections (some error with GZipStream in .net Core).

**Comment by angelocart** [comment]
Thank you for the response! I will let you know if any other errors occur.

Does the previous documentation for xNet apply to the new repository?

**Comment by Fedorus** [comment]
> Thank you for the response! I will let you know if any other errors occur.
> 
> Does the previous documentation for xNet apply to the new repository?

There was no breakink changes in API, but there may be bugs in fixes that I`ve described above

**Comment by angelocart** [comment]
> > Thank you for the response! I will let you know if any other errors occur.
> > Does the previous documentation for xNet apply to the new repository?
> 
> There was no breakink changes in API, but there may be bugs in fixes that I`ve described above

Okay. Thank you again for the quick reply.

**Comment by angelocart** [comment]
@Fedorus I use Visual Studio 2015. I cannot open the project files for xNetStandart so I cannot build the DLL file. Could you build the DLL and upload it?

**Comment by Fedorus** [comment]
> @Fedorus I use Visual Studio 2015. I cannot open the project files for xNetStandart so I cannot build the DLL file. Could you build the DLL and upload it?

here
[xNetStandart.zip](https://github.com/X-rus/xNet/files/2812887/xNetStandart.zip)

**Comment by angelocart** [comment]
> > @Fedorus I use Visual Studio 2015. I cannot open the project files for xNetStandart so I cannot build the DLL file. Could you build the DLL and upload it?
> 
> here
> [xNetStandart.zip](https://github.com/X-rus/xNet/files/2812887/xNetStandart.zip)

Much appreciated. Have a good day!

**Comment by AnErrupTion** [comment]
**This issue has been fixed with Better-xNet.** https://github.com/AnErrupTion/Better-xNet @xerasy

**Comment by mathewsun** [comment]
> Replacing the line `sslStream.AuthenticateAsClient(...)` **with** `sslStream.AuthenticateAsClient(address.Host, null, SslProtocols.Tls12, false);` returns an error of "SslProtocols' does not contain a definition for 'Tls12'.
> 
> Please fix, thanks!

sslStream.AuthenticateAsClient(address.Host, null, (SslProtocols)3072, true);

**Comment by soheil-js** [comment]
This issue has been fixed with MVNet.
Supports TLS / SSL
https://github.com/SoheilMV/MVNet
